File: server/internal/model/school.go
// server/internal/model/school.go
package model

// CreateSchoolRequest 定义了创建学校时的请求体
type CreateSchoolRequest struct {
	Name          string `json:"name" binding:"required"`
	ContactName   string `json:"contactName"`
	ContactPhone  string `json:"contactPhone"`
	Address       string `json:"address"`
	AdminUsername string `json:"adminUsername" binding:"required"`
	AdminPassword string `json:"adminPassword" binding:"required"`
	AdminRealName string `json:"adminRealName" binding:"required"`
}

// UpdateSchoolRequest 定义了更新学校时的请求体
type UpdateSchoolRequest struct {
	Name         string `json:"name" binding:"required"`
	ContactName  string `json:"contactName"`
	ContactPhone string `json:"contactPhone"`
	Address      string `json:"address"`
	IsEnabled    bool   `json:"isEnabled"`
}

// SchoolListItem 定义了学校列表返回的结构
type SchoolListItem struct {
	SysOrganization
	AdminUsername string `json:"adminUsername"`
}

File: web/src/api/school.ts
// web/src/api/school.ts
import apiClient from './index';

export interface School {
  ID: number;
  Name: string;
  OrgType: number;
  ContactName: string;
  ContactPhone: string;
  Address: string;
  IsEnabled: boolean;
  AdminUserID: number;
  AdminUsername: string;
  CreatedAt: string;
  UpdatedAt: string;
}

export interface CreateSchoolPayload {
  name: string;
  contactName: string;
  contactPhone: string;
  address: string;
  adminUsername: string;
  adminPassword: string;
  adminRealName: string;
}

export interface UpdateSchoolPayload {
  Name: string;
  ContactName: string;
  ContactPhone: string;
  Address: string;
  IsEnabled: boolean;
}

export interface ListSchoolsResponse {
  list: School[];
  total: number;
}

/**
 * 获取学校/站点列表
 */
export function listSchoolsApi(params: {
  page: number;
  pageSize: number;
}): Promise<ListSchoolsResponse> {
  return apiClient.get('/schools', { params });
}

/**
 * 创建一个新的学校/站点
 */
export function createSchoolApi(data: CreateSchoolPayload): Promise<any> {
  return apiClient.post('/schools', data);
}

/**
 * 更新一个学校/站点的信息
 */
export function updateSchoolApi(id: number, data: UpdateSchoolPayload): Promise<any> {
  return apiClient.put(`/schools/${id}`, data);
}

/**
 * 删除一个学校/站点
 */
export function deleteSchoolApi(id: number): Promise<any> {
  return apiClient.delete(`/schools/${id}`);
}
